### PR TITLE
Add config script & scheduled trigger

### PR DESCRIPTION
**Pull request by sunmou99**
Add config script and use the script in workflow.
Add scheduled trigger.

Not in this PR:
no config for other workflows except integration test workflow.
some configs are not used for now, but will be used in the future.
auto-diff feature.

### CODE REVIEW

File: scripts/gha/print_matrix_configuration.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+      "android_device": ["android_target", "emulator_target"],
+      "ios_device": ["ios_target", "simulator_target"],
+
+      MINIMAL_KEY: {
+        "platform": ["Desktop"],
+        "apis": "firestore"
+      },
+
+      EXPANDED_KEY: {
+        # "2019.2.8f1", "2021.1.9f1" haven't been varified.
```
Comment (yifanyang) [inline scripts/gha/print_matrix_configuration.py:65]:
```suggestion
        # "2019.2.8f1", "2021.1.9f1" haven't been verified.
```

File: scripts/gha/print_matrix_configuration.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+    "config": {
+      "platform": "Windows,macOS,Linux,Android,iOS",
+      "apis": "analytics,auth,crashlytics,database,dynamic_links,functions,installations,messaging,remote_config,storage",
+      "mobile_test_on": "real,virtual"
+    }
+  },
+}
+
+TEST_DEVICES = {
+  "android_min": {"type": "real", "model":"Nexus10", "version":"19"},
```
Comment (yifanyang) [inline scripts/gha/print_matrix_configuration.py:80]:
```suggestion
  "android_min": {"type": "real", "model": "Nexus10", "version": "19"},
```

File: scripts/gha/print_matrix_configuration.py [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+  # '["2017.4.37f1", "2019.2.8f1", "2021.1.9f1"]'
+
+  # Eg: for strings
+  # print(json.dumps) -> "flame"
+  # print(repr(json.dumps)) -> '"flame"'
+
+  print(json.dumps(value))
+
+
+### TODO
```
Comment (yifanyang) [inline scripts/gha/print_matrix_configuration.py:154]: [go/todo-style](http://go/todo-style)

**Review by yifanyang — DISMISSED** [review]

File: scripts/gha/print_matrix_configuration.py (hunk 2 above)
Comment (yifanyang) [reply scripts/gha/print_matrix_configuration.py:80]: Same to the following lines.

**Review by jonsimantov — APPROVED** [review]

**Pull request merged**